crates/tx-graph/src/connectors/connector_s.rs

```diff
@@ -47,3 +54,410 @@ impl ConnectorS {
         finalize_input(input, [signature.as_ref()]);
     }
 }
+
+/// The connector to move the operator's stake across Stake transactions.
+///
+/// It is used in the Disprove and Slash Stake `k` transactions, where `k` is the index of the
+/// stake transaction.
+///
+/// The operator can also advance the stake chain by revealing the pre-image, along with a valid
+/// signature from the operator's public key.
+/// Note that the stake advancement is done by the `stake-chain` crate.
+///
+/// To illustrate the concept, let's say that an operator wants to claim the `k`th bridged-in UTXO.
+/// For this, they need the `k`th Claim Transaction and hence the `k`th stake transaction. An
+/// operator is not required to periodically advance the stake chain, so it may be the case that
+/// they have only posted the `k-n`th stake chain. In this case, they post the next `n` stake
+/// transactions at an interval of `ΔS`. We can set the value of `ΔS` to a small enough value while
+/// still preventing an operator from spamming the system with faulty claims. Once the chain has
+/// been advanced, they can use the `k`th stake to make their claim.
+///
+/// If the operator tries to advance the chain to the `k+1`th stake, they are required to reveal a
+/// pre-image (publicly) on bitcoin. Using this stake, anybody can post the Burn Payouts transaction
+/// which renders it impossible for an operator to receive a payout (optimistically or otherwise).
+///
+/// If the operator has received a `k`th Payout Optimistic or Payout transaction, they can advance
+/// the stake chain (revealing the pre-image) without fear. It is the responsibility of the
+/// [`ConnectorP`](super::connector_p::ConnectorP) and [`ConnectorStake`] to ensure that will make
+/// it impossible make it impossible
+///
+/// # Security
+///
+/// An operator can only advance the stake chain if they reveal the pre-image along with a valid
+/// signature from the operator's public key. Hence, the operator must must be able to provide the
+/// pre-image to the [`ConnectorStake`]. It is advised to use a secure storage mechanism to store
+/// the pre-image, and optionally a deterministic key derivation function to derive the pre-image
+/// from something such as an operator's master entropy.
+// TODO: This should replace the `ConnectorS` struct above.
+#[derive(Debug, Clone, Copy)]
+pub struct ConnectorStake {
+    /// The N-of-N aggregated public key for the operator set.
+    n_of_n_agg_pubkey: XOnlyPublicKey,
+
+    /// The operator's public key.
+    operator_pubkey: XOnlyPublicKey,
+
+    /// The corresponding pre-image for the `k`th stake.
+    ///
+    /// This is a 32-byte array, that should be generated securely from the operator's master
+    /// entropy along with the `k`th stake's index.
+    ///
+    /// A suggestion is to hash them together, i.e. SHA256(master entropy || stake_index), and take
+    /// the 32 bytes of the resulting hash.
```

Comment:
See related comment.

Reply:
324ec8d1dfc8e7ac80a10f93239ad9b084d13b4a, see if that solves it

Reply:
See related comment!

Reply:
e24f7398b1f43f1644a8ddb3a131ca46af492f59

Reply:
Gonna resolve for now, let me know if this is not appropriate.